Sources/Foundation/Process.swift

```diff
@@ -939,21 +939,11 @@ open class Process: NSObject, @unchecked Sendable {
             useFallbackChdir = false
         }
 
-#if canImport(Darwin) || os(Android) || os(OpenBSD) || os(FreeBSD)
-        var spawnAttrs: posix_spawnattr_t? = nil
-#else
-        var spawnAttrs: posix_spawnattr_t = posix_spawnattr_t()
-#endif
-        try _throwIfPosixError(posix_spawnattr_init(&spawnAttrs))
-#if os(Android)
-        guard var spawnAttrs else {
-            throw NSError(domain: NSPOSIXErrorDomain, code: Int(errno),
-                          userInfo: [NSURLErrorKey:self.executableURL!])
-        }
-#endif
-        try _throwIfPosixError(posix_spawnattr_setflags(&spawnAttrs, .init(POSIX_SPAWN_SETPGROUP)))
+        let spawnAttrs = _CFPosixSpawnAttrAlloc()
+        try _throwIfPosixError(_CFPosixSpawnAttrInit(spawnAttrs))
```

Comment:
I would recommend adding `defer` here, and cleaning up in one place. A lot of intermediate `_throwIfPosixError` are not cleaning up memory for these attributes.